src-java/blue-green/src/main/java/org/openkilda/bluegreen/kafka/interceptors/VersioningProducerInterceptor.java

```diff
@@ -0,0 +1,96 @@
+/* Copyright 2020 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.bluegreen.kafka.interceptors;
+
+import static org.openkilda.bluegreen.kafka.Utils.MESSAGE_VERSION_HEADER;
+import static org.openkilda.bluegreen.kafka.Utils.PRODUCER_COMPONENT_NAME_PROPERTY;
+import static org.openkilda.bluegreen.kafka.Utils.PRODUCER_RUN_ID_PROPERTY;
+import static org.openkilda.bluegreen.kafka.Utils.PRODUCER_ZOOKEEPER_CONNECTION_STRING_PROPERTY;
+import static org.openkilda.bluegreen.kafka.Utils.getValue;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.clients.producer.ProducerInterceptor;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+
+import java.time.Instant;
+import java.util.Map;
+
+@Slf4j
+public class VersioningProducerInterceptor<K, V> extends VersioningInterceptorBase
+        implements ProducerInterceptor<K, V> {
+
+    public VersioningProducerInterceptor() {
+        log.debug("Initializing VersioningProducerInterceptor");
+    }
+
+    @Override
+    public ProducerRecord<K, V> onSend(ProducerRecord<K, V> record) {
+        if (watchDog == null) {
+            // try to reconnect
+            initWatchDog();
+        }
+
+        if (record.headers().headers(MESSAGE_VERSION_HEADER).iterator().hasNext()) {
+            log.info("Kafka record {} already has header {}: {}",
+                    record, MESSAGE_VERSION_HEADER,
+                    new String(record.headers().headers(MESSAGE_VERSION_HEADER).iterator().next().value()));
+            record.headers().remove(MESSAGE_VERSION_HEADER);
+        }
+
+        if (version == null) {
+            if (isVersionIsNotTimeoutPassed()) {
```

Comment:
As for me it should be in base class. We should have method `warnBuildVersionNotSet()`, which one will incapsulate log each N seconds logic.

PS As for me we should calc and log the amount of "skipped" records.

Reply:
Unfortunately we have different logs and different logic for such case in producer and consumer. So we cant move it into base class